orchestrator/careplanservice/handle_taskfiller.go

```diff
@@ -0,0 +1,379 @@
+	"github.com/SanteonNL/orca/orchestrator/lib/to"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"github.com/samply/golang-fhir-models/fhir-models/fhir"
+)
+
+var primaryTaskFocusToQuestionnaireURL = map[string]string{
+	"2.16.528.1.1007.3.3.21514.ehr.orders|99534756439": "http://decor.nictiz.nl/fhir/Questionnaire/2.16.840.1.113883.2.4.3.11.60.909.26.34-1--20240902134017",
+}
+
+var followUpQuestionnaireMap = map[string]string{
+	"http://decor.nictiz.nl/fhir/Questionnaire/2.16.840.1.113883.2.4.3.11.60.909.26.34-1--20240902134017": "http://decor.nictiz.nl/fhir/Questionnaire/2.16.840.1.113883.2.4.3.11.60.909.26.34-2--20240902134017",
+}
+
+func (s *Service) handleTaskFillerCreate(task *fhir.Task) error {
+	log.Info().Msg("Running handleTaskFillerCreate")
+
+	if !s.isScpTask(task) {
+		log.Info().Msg("Task is not an SCP Task - skipping")
+		return nil
+	}
+
+	if err := s.isValidTask(task); err != nil {
+		log.Error().Msgf("Task invalid - skipping: %v", err)
+		return fmt.Errorf("task is not valid - skipping: %w", err)
+	}
+
+	partOfRef, err := s.partOf(task)
+	if err != nil {
+		return fmt.Errorf("failed to extract Task.partOf: %w", err)
+	}
+
+	// If partOfRef is nil, handle the task as a primary task - no need to create follow-up subtasks for newly created Tasks
+	//This only happens on Task update where the Task.output is filled with a QuestionnaireResponse
+	if partOfRef == nil {
+		log.Info().Msg("Found a new 'primary' task, checking if more information is needed via a Questionnaire")
+		err := s.createSubTaskOrFinishPrimaryTask(task, true)
+		if err != nil {
+			return fmt.Errorf("failed to process new primary Task: %w", err)
+		}
+	}
+	return nil
+}
+
+func (s *Service) handleTaskFillerUpdate(task *fhir.Task) error {
+
+	log.Info().Msg("Running handleTaskFillerUpdate")
+
+	if !s.isScpTask(task) {
+		log.Info().Msg("Task is not an SCP Task - skipping")
+		return nil
+	}
+
+	if task.Status != fhir.TaskStatusCompleted {
+		log.Info().Msg("Task.status is not completed - skipping")
+		return nil
+	}
+
+	if err := s.isValidTask(task); err != nil {
+		log.Error().Msgf("Task invalid - skipping: %v", err)
+		return fmt.Errorf("task is not valid - skipping: %w", err)
+	}
+
+	partOfRef, err := s.partOf(task)
+	if err != nil {
+		return fmt.Errorf("failed to extract Task.partOf: %w", err)
+	}
+
+	if partOfRef == nil {
+		return errors.New("handleTaskFillerUpdate got a subtask without a partOf set")
+	}
+
+	log.Info().Msg("SubTask.status is completed - processing")
+
+	return s.createSubTaskOrFinishPrimaryTask(task, false)
+
+}
+
+func (s *Service) markPrimaryTaskAsCompleted(subTask *fhir.Task) error {
+	log.Info().Msg("Marking primary Task as completed")
+
+	if subTask.PartOf == nil {
+		return errors.New("task.partOf is empty")
+	}
+
+	partOfTaskRef := subTask.PartOf[0].Reference
+	if partOfTaskRef == nil || !strings.HasPrefix(*partOfTaskRef, "Task/") {
+		return errors.New("task.partOf[0].reference is not a valid Task reference")
+	}
```

Comment:
I think you made a util function for this, `partOf()` (although you need to handle the `nil` return)